app.js

```diff
@@ -899,3 +1051,31 @@ function fail(s) {
   console.error(s);
   process.exit(1);
 }
+
+function nameToLiteralOrIdentifier(name) {
+  if (name.value.computed) {
+    return false;
+  }
+  if (name.value.type === 'Identifier') {
+    return name.value;
+  }
+  if (name.value.type === 'Literal') {
+    // Where possible convert to identifier
+    const text = name.value.value;
+    if (text.match(/^[a-zA-Z]\w*$/)) {
+      return {
+        type: 'Identifier',
+        name: text
+      };
+    } else {
+      return name.value;
+    }
+  }
+  return false;
+}
+
+function unsupported() {
+  const e = new Error('Unsupported');
+  e.name = 'unsupported';
```

Comment:
Yep, the name property is a documented thing. javascript errors are funny beasts.